src/iterator.c

```diff
@@ -555,8 +555,33 @@ void qd_iterator_annotate_space(qd_iterator_t *iter, const char* space, int spac
 
 unsigned char qd_iterator_octet(qd_iterator_t *iter)
 {
-    if (!iter)
-        return 0;
+    assert(iter);
```

Comment:
It is good to add this assert but can we still keep this line ?
    if (!iter)
        return 0;

It saves the router from a crash in production environment

Reply:
yeah, was hoping to save a bit there, but look like the rest of the api verifies the iter pointer, so I'll add it back